src/llvm/basic_block.cr

```diff
@@ -17,4 +17,9 @@ struct LLVM::BasicBlock
   def to_unsafe
     @unwrap
   end
+
+  def name
+    block_name = LibLLVMExt.basic_block_name(self)
+    block_name ? LLVM.string_and_dispose(block_name) : "unknown"
```

Comment:
Since a name is optional, I would expect this to return `String?`.

Reply:
(the "unknown" can be inserted on the caller site, like `block.name || "unknown"`)

Reply:
I'm not sure names are optional. In llvm < 4.0 we might not be able to know the name, but basic blocks always have a name I think. So I was catching that and ensuring to return a `String`.

If you think is still better to return a `String?` I can apply the suggested changes.

Reply:
Yes, basic blocks can have no name, for example the entry block is usually left without a name. In the API for creating blocks the default name is `""` if not given. So I think we should either return `nil` or `""`, but not "undefined" (because there could be a block named like that).

Reply:
Ok. done!